java/flight/flight-grpc/src/test/java/org/apache/arrow/flight/TestFlightGrpcUtils.java

```diff
@@ -85,6 +86,42 @@ public void testMultipleGrpcServices() throws IOException {
     Assert.assertEquals(Empty.newBuilder().build(), blockingStub.test(Empty.newBuilder().build()));
   }
 
+  @Test
+  public void testShutdown() throws IOException, InterruptedException {
+
+    //Defines flight service
+    final BufferAllocator allocator = new RootAllocator(Integer.MAX_VALUE);
+    final NoOpFlightProducer producer = new NoOpFlightProducer();
+    final ServerAuthHandler authHandler = ServerAuthHandler.NO_OP;
+    final ExecutorService exec = Executors.newCachedThreadPool();
+    final BindableService flightBindingService = FlightGrpcUtils.createFlightService(allocator, producer,
+        authHandler, exec);
+
+    //initializes server with 2 services - FlightBindingService & TestService
+    final String serverName = InProcessServerBuilder.generateName();
+    final Server server = InProcessServerBuilder.forName(serverName)
+        .directExecutor()
+        .addService(flightBindingService)
+        .addService(new TestServiceAdapter())
+        .build();
+    server.start();
+
+    //Initializes channel so that multiple clients can communicate with server
+    final ManagedChannel managedChannel = InProcessChannelBuilder.forName(serverName)
```

Comment:
try-with-resources? Otherwise we need to explicitly close this channel, since it now gets proxied

Reply:
ManagedChannel is not AutoCloseable, so we can't do that.